sim/dex-data.ts

```diff
@@ -90,105 +92,82 @@ export class BasicEffect {
 	 * necessarily the current gen being simulated.) Not all effects
 	 * track generation; this will be 0 if not known.
 	 */
-	gen: number;
+	gen: number; // TODO: Generation
```

Comment:
Probably out of scope for this change, but `type Generation = 1 | 2 | 3 | 4 | 5 | 6 | 7`? Not sure there's value, but it seems like something which would be an enum in most languages.

Reply:
I'm neutral on that change.